packages/ckb/src/collector/offline.ts

```diff
@@ -0,0 +1,91 @@
+import { IndexerCell } from '../types/collector';
+import { Collector } from './index';
+import { isScriptEqual } from '../utils/ckb-tx';
+import { Hex } from '../types';
+
+export class OfflineCollector extends Collector {
+  private cells: IndexerCell[];
+
+  constructor(cells: IndexerCell[]) {
+    super({ ckbNodeUrl: '', ckbIndexerUrl: '' });
+    this.cells = cells;
+  }
+
+  getCkb(): never {
+    throw new Error('OfflineCollector does not have a CKB instance');
+  }
+
+  async getCells({
+    lock,
+    type,
+    isDataMustBeEmpty = true,
+    outputCapacityRange,
+  }: {
+    lock?: CKBComponents.Script;
+    type?: CKBComponents.Script;
+    isDataMustBeEmpty?: boolean;
+    outputCapacityRange?: Hex[];
+  }): Promise<IndexerCell[]> {
+    let cells: IndexerCell[] = [];
+
+    if (lock) {
+      cells = this.cells.filter((cell) => isScriptEqual(cell.output.lock, lock));
+    } else if (type) {
+      cells = this.cells.filter((cell) => {
+        if (!cell.output.type) {
+          return false;
+        }
+        return isScriptEqual(cell.output.type, type);
+      });
+    }
```

Comment:
The filter does not meet expectations if the lock and type are null.

Reply:
Fixed it in a3d533d4d735d323374f09fce0a27ab714f955bd.